lib/ssl/test/ssl_sni_SUITE.erl

```diff
@@ -186,15 +190,38 @@ sni_no_match(Config) ->
     ClientOptions =  ssl_test_lib:ssl_options([{server_name_indication, HostName} |
                                                proplists:get_value(client_opts, Config)], Config),
     ServerOptions =  ssl_test_lib:ssl_options(DefaultConf, Config),
-    basic_sni_alert_test(ServerNode, ServerOptions, ClientNode, ClientOptions, HostName).
+    basic_sni_alert_test(ServerNode, ServerOptions, ClientNode, ClientOptions, HostName, handshake_failure).
 
 sni_no_match_fun(Config) ->
     {ClientNode, ServerNode, HostName} = ssl_test_lib:run_where(Config),
-    [{sni_hosts, _}| DefaultConf] = proplists:get_value(sni_server_opts, Config),
-    ServerOptions =  ssl_test_lib:ssl_options(DefaultConf, Config),
-    ClientOptions =  ssl_test_lib:ssl_options([{server_name_indication, HostName} |
+    [{sni_hosts, ServerSNIConf}| DefaultConf] = proplists:get_value(sni_server_opts, Config),
+    SNIFun = fun(Domain) -> proplists:get_value(Domain, ServerSNIConf, unrecognized) end,
+    ServerOptions =  ssl_test_lib:ssl_options(DefaultConf, Config) ++ [{sni_fun, SNIFun}],
+    ClientOptions =  ssl_test_lib:ssl_options([{server_name_indication, "localhost"} |
                                                proplists:get_value(client_local_opts, Config)], Config),
-    basic_sni_alert_test(ServerNode, ServerOptions, ClientNode, ClientOptions, HostName).
+    basic_sni_alert_test(ServerNode, ServerOptions, ClientNode, ClientOptions, HostName, unrecognized_name).
+
+sni_fail_fun(Config) ->
+    {ClientNode, ServerNode, HostName} = ssl_test_lib:run_where(Config),
+    [_| DefaultConf] = proplists:get_value(sni_server_opts, Config),
+    ServerOptions =  ssl_test_lib:ssl_options(DefaultConf, Config) ,
+    ClientOptions =  ssl_test_lib:ssl_options([{server_name_indication, HostName} |
+                                               proplists:get_value(client_opts, Config)], Config),
+    basic_sni_alert_test(ServerNode, ServerOptions ++ [{sni_fun, fun(_Domain) -> [{versions, ['tlsv1.5']}] end}],
```

Comment:
what is the intent behind `tlsv1.5` value?
maybe looks to good if it is supposed to be be interpretated as wrong unsupported version.

Reply:
The intent is that it should be an invalid option